### PR TITLE
adds missing header, removes Bazel unnecessary dependency

### PR DESCRIPTION
**Pull request by cjdb**

### CODE REVIEW

File: utils/bazel/llvm-project-overlay/mlir/BUILD.bazel [hunk 1]
```diff
@@ -11719,6 +11719,25 @@ gentbl_cc_library(
...
+    deps = [
+        ":ArithDialect",
+        ":BytecodeOpInterface",
+        ":MaskableOpInterface",
+        ":VectorAttributesIncGen",
+        ":VectorDialectIncGen",
+        ":VectorOpsIncGen",
+    ],
+)
+
```
Comment (cjdb) [inline utils/bazel/llvm-project-overlay/mlir/BUILD.bazel]:
```suggestion
```

File: mlir/include/mlir/Dialect/Affine/IR/ValueBoundsOpInterfaceImpl.h [hunk 2]
```diff
@@ -9,6 +9,7 @@
 #ifndef MLIR_DIALECT_AFFINE_IR_VALUEBOUNDSOPINTERFACEIMPL_H
 #define MLIR_DIALECT_AFFINE_IR_VALUEBOUNDSOPINTERFACEIMPL_H
 
+#include <cstdint>
```
Comment (rupprecht) [inline mlir/include/mlir/Dialect/Affine/IR/ValueBoundsOpInterfaceImpl.h:12]: nit: formatting

**Review by rupprecht — COMMENTED** [review]
MLIR seems to be passing at trunk, at least on buildkite. What issue does this fix?

I would expect the cstdint to fix a CppHeaderCompile issue (being able to compile a standalone header), and adding the MaskableOpInterface would fix a layering check. Why is VectorOpsIncGen being removed though?

Otherwise, this kind of change seems fine. LGTM with a tiny bit of context added to the PR description.

**Comment by cjdb** [comment]
> MLIR seems to be passing at trunk, at least on buildkite. What issue does this fix?
> 
> I would expect the cstdint to fix a CppHeaderCompile issue (being able to compile a standalone header), and adding the MaskableOpInterface would fix a layering check. Why is VectorOpsIncGen being removed though?
> 
> Otherwise, this kind of change seems fine. LGTM with a tiny bit of context added to the PR description.

Bazel isn't happy without this change. Specifically:

```
$ bazel build --config=generic_clang @llvm-project//mlir/test/Analysis:DataFlow/test-dead-code-analysis.mlir.test
INFO: Analyzed target @@llvm-project//mlir/test/Analysis:DataFlow/test-dead-code-analysis.mlir.test (1 packages loaded, 4265 targets configured).
ERROR: /home/cjdb/.cache/bazel/_bazel_cjdb/7630592ec0cf69943691bbf5cc2fc71f/external/llvm-project/mlir/BUILD.bazel:3913:11: Compiling mlir/lib/Dialect/Affine/IR/ValueBoundsOpInterfaceImpl.cpp failed: (Exit 1): clang-20 failed: error executing CppCompile command (from target @@llvm-project//mlir:AffineDialect) /home/cjdb/opt/bin/clang-20 -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -Wunused-but-set-parameter -Wno-free-nonheap-object -fcolor-diagnostics ... (remaining 124 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
In file included from external/llvm-project/mlir/lib/Dialect/Affine/IR/ValueBoundsOpInterfaceImpl.cpp:9:
external/llvm-project/mlir/include/mlir/Dialect/Affine/IR/ValueBoundsOpInterfaceImpl.h:31:11: error: unknown type name 'int64_t'; did you mean '__int64_t'?
   31 | FailureOr<int64_t> fullyComposeAndComputeConstantDelta(Value value1,
      |           ^
/usr/include/x86_64-linux-gnu/bits/types.h:44:25: note: '__int64_t' declared here
   44 | typedef signed long int __int64_t;
      |                         ^
1 error generated.
Target @@llvm-project//mlir/test/Analysis:DataFlow/test-dead-code-analysis.mlir.test failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 6.603s, Critical Path: 3.73s
INFO: 17 processes: 13 internal, 4 linux-sandbox.
ERROR: Build did NOT complete successfully
```

**Pull request merged**